.github/workflows/jacktrip.yml

```diff
@@ -597,13 +580,105 @@ jobs:
           path: ${{ env.BUILD_PATH }}/${{ matrix.bundle-path }}
       - name: upload installer
         uses: actions/upload-artifact@v2
-        if: matrix.installer-path && (!startsWith(github.ref, 'refs/tags/') || (startsWith(github.ref, 'refs/tags/') && matrix.release-name)) && steps.check-secrets.outputs.available != 'true'
+        if: matrix.installer-path && (!startsWith(github.ref, 'refs/tags/') || (startsWith(github.ref, 'refs/tags/') && matrix.release-name)) && (steps.check-secrets.outputs.available != 'true' || runner.os == 'Windows')
+        with:
+          name: JackTrip-${{ steps.set-version.outputs.version }}-${{ steps.set-version.outputs.name }}-installer
+          path: ${{ env.BUILD_PATH }}/${{ matrix.installer-path }}
+  check-secrets:
+    name: Check if secrets exist to determine if build & sign run
+    runs-on: ubuntu-latest
+    outputs:
+      should_run: ${{ steps.setvar.outputs.should_run }}
+    steps:
+      - id: setvar
+        run: |
+          if [[ "${{ secrets.APP_SIGNING_CERT_BASE64 }}" != "" && \
+                "${{ secrets.INSTALLER_SIGNING_CERT_BASE64 }}" != "" && \
+                "${{ secrets.CERT_PASSPHRASE }}" != "" && \
+                "${{ secrets.APP_CERT_NAME }}" != "" && \
+                "${{ secrets.INSTALLER_CERT_NAME }}" != "" && \
+                "${{ secrets.APPLE_APP_SPECIFIC_PWD }}" != "" && \
+                "${{ secrets.TOTP_SECRET }}" != "" && \
+                "${{ secrets.ESIGNER_CREDENTIAL_ID }}" != "" && \
+                "${{ secrets.SSL_COM_PWD }}" != "" && \
+                "${{ secrets.KEYCHAIN_PWD }}" != "" ]]; \
+          then
+            echo "Secrets to sign artifacts were configured in the repo"
+            echo "should_run=true" >> $GITHUB_OUTPUT
+          else
+            echo "Secrets to sign artifacts were not configured in the repo"
+            echo "should_run=false" >> $GITHUB_OUTPUT
+          fi
+  sign:
+    needs: [build, check-secrets]
+    if: needs.check-secrets.outputs.should_run == 'true'
```

Comment:
I think you might not need the `check-secrets` job at all. Would it work to use this conditional here?
```
if: needs.build.steps.check-secrets.outputs.available == 'true'
```
or something like that - basically use the steps that already run in the main build job.

Reply:
also, it should only run if we have the installer to sign, so maybe additional condition similar to what we had for signing step previously; or just run on tagged builds

Reply:
I basically solved this by replacing the step with the check-secrets job

Reply:
Honestly, just because I find it easier to debug things. Hiding it in the step doesn't give me a green check mark on if it passes or not and since it's the first thing that runs, i don't have to wait 10 minutes for a build.

Reply:
okay, that makes sense.